File: api/src/main/kotlin/org/team619/rutor/core/Converter.java
package org.team619.rutor.core;

import org.jsoup.nodes.Element;

/**
 * Created by BORIS on 07.08.2016.
 */
public interface Converter<OUTPUT, INPUT> {

    OUTPUT convert(INPUT input);

    default boolean support(Element element) {
        return Boolean.FALSE;
    }

    default int priority() {
        return 1;
    }

}

File: api/src/main/kotlin/org/team619/rutor/core/Converter.kt
package org.team619.rutor.core

import org.jsoup.nodes.Element

/**
 * Created by BORIS on 07.08.2016.
 */
interface Converter<OUTPUT, INPUT> {

    fun convert(input: INPUT): OUTPUT

    fun support(element: Element): Boolean {
        return java.lang.Boolean.FALSE
    }

    fun priority(): Int {
        return 1
    }

}
